datalad/__init__.py

```diff
@@ -16,6 +16,11 @@
 from .config import ConfigManager
 cfg = ConfigManager()
 
+from datalad.support.sshconnector import SSHManager
```

Comment:
singleton or not is somewhat orthogonal.  I was just asking to move construction of the beast until later, i.e. when finally needed.  It could as well reside under `datalad/support/ssh.py` so those submodules which need it, would import it from there, thus delaying its construction until it is needed.
It is all a minor cost, but it all adds up